Create a Dashboard for logged in users and refresh the webUI

ryanlerch opened the pull request:
This re-arranges the webUI a little bit to make way for a
dashboard page for logged in users, that shows their active
updates and overrides. When logged in, the home page looks like this:

![screencapture-0-0-0-0-6543-2019-09-12-15_58_29](https://user-images.githubusercontent.com/592259/64757812-79554780-d576-11e9-90cf-98c36881c2a9.png)

and when not logged in, like this:

![screencapture-0-0-0-0-6543-2019-09-12-16_00_12](https://user-images.githubusercontent.com/592259/64757809-78bcb100-d576-11e9-8cef-083bd8f41467.png)

The updates list page is also refreshed, looking like this:
![screencapture-0-0-0-0-6543-updates-2019-09-12-15_59_31](https://user-images.githubusercontent.com/592259/64757810-78bcb100-d576-11e9-9504-7db6122bcb18.png)

And all the counts on the front page are now moved to the Releases page, and a releases menuitem now in the top menu:
![screencapture-0-0-0-0-6543-releases-2019-09-12-15_58_57](https://user-images.githubusercontent.com/592259/64757811-78bcb100-d576-11e9-85bf-281a4ce7e8a7.png)

The actual page for an update has also been refreshed, removing a lot of the colours that were used for everything, with colour only really used for karma, and the status of the update itself.

![screencapture-0-0-0-0-6543-updates-FEDORA-2019-095d8600a1-2019-09-12-16_13_14](https://user-images.githubusercontent.com/592259/64758578-4318c780-d578-11e9-9744-3631d24ce9ad.png)

ryanlerch committed:
Create a Dashboard for logged in users and refresh the webUI

This re-arranges the webUI a little bit to make way for a
dashboard page for logged in users, that shows their active
updates and overrides.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

pypingou commented:
The screenshots are really looking nice :)

Does this address: https://github.com/fedora-infra/bodhi/issues/3429 and https://github.com/fedora-infra/bodhi/issues/3424 ?

ryanlerch commented:
> The screenshots are really looking nice :)
> 
> Does this address: #3429 and #3424 ?

#3429 Definitely does fix that one.

#3424 somewhat. It makes the karma stuff not as in your face, so i think just turning off autokarma when autofilining automatic updates might be enough to finish off 3424.

Zlopez commented on bodhi/server/static/js/site.js:
```diff
@@ -39,6 +39,8 @@ $(document).ready(function() {
 
     // Attach bootstrap tooltips to everything.
     $('[data-toggle="tooltip"]').tooltip();
+    $('[data-toggle="popover"]').popover();
```
Is this another type of tooltip?

ryanlerch replied:
kinda, it's another bootstrap component that is kinda inbetween a tooltip and a modal. I'm using it here for the little "help" popup for comment edting.

https://getbootstrap.com/docs/4.3/components/popovers/

Zlopez commented on bodhi/server/views/generic.py:
```diff
@@ -87,123 +117,66 @@ def get_latest_updates(critpath, security):
     Generate and return a dictionary of stats for the home() function to use.
 
-    This function returns a dictionary with the following 4 keys:
+    This function returns a dictionary with the following 3 keys:
```
I see 4 keys

ryanlerch replied:
there are actually 5 now, updated the docblock, adding the description for top_testers as well.

Zlopez reviewed (commented):
Few comments

Zlopez commented:
Did you updated jquery?

I see this in web console, when running this PR:
```
Source map error: request failed with status 404
Resource URL: http://localhost:6543/static/v4.1.1/js/jquery-1.10.2.min.js
Source Map URL: jquery-1.10.2.min.map
```

I also see this:
```
Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery-1.10.2.min.js:2:21
```

Zlopez commented:
Locally I don't see anything in Active testers and Active packagers categories, not sure if I should see anything.
![Screenshot from 2019-09-12 13-35-33](https://user-images.githubusercontent.com/6943409/64781074-5751ca00-d562-11e9-85b7-218b25be160e.png)

Zlopez commented:
I must say, I really like how release page looks right now. The graph on the top is just amazing :-)
![Screenshot from 2019-09-12 13-37-47](https://user-images.githubusercontent.com/6943409/64781201-9aac3880-d562-11e9-94d6-246ca52440f2.png)

Zlopez commented:
Found one small issue, when adding karma to update. Karma counter is not changed until refresh.
![Screenshot from 2019-09-12 13-42-59](https://user-images.githubusercontent.com/6943409/64781533-59685880-d563-11e9-9b22-1c54b8c18c6b.png)

ryanlerch commented:
> Did you updated jquery?
> 
> I see this in web console, when running this PR:
> 
> ```
> Source map error: request failed with status 404
> Resource URL: http://localhost:6543/static/v4.1.1/js/jquery-1.10.2.min.js
> Source Map URL: jquery-1.10.2.min.map
> ```
> 
> I also see this:
> 
> ```
> Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery-1.10.2.min.js:2:21
> ```

hmmm, i didn't touch jquery, and not getting the warning here either. Does the warning also come up when running develop?

ryanlerch commented:
> Locally I don't see anything in Active testers and Active packagers categories, not sure if I should see anything.
> ![Screenshot from 2019-09-12 13-35-33](https://user-images.githubusercontent.com/6943409/64781074-5751ca00-d562-11e9-85b7-218b25be160e.png)

Yeah, this is showing the last 7 days of activity. (this is the way it currently works in prod.) my guess is that the db on your devel box is over 7 days old, so no activity has happened. with live bodhi, i very much assume that we will not have a 7 day period with no activity, so this should always be populated on prod.

ryanlerch commented:
> I must say, I really like how release page looks right now. The graph on the top is just amazing :-)
> ![Screenshot from 2019-09-12 13-37-47](https://user-images.githubusercontent.com/6943409/64781201-9aac3880-d562-11e9-94d6-246ca52440f2.png)

This is actually a page that i didnt really touch. The graph has been there for a few years :D -- just getting to the release page was a bit hidden before. However, this page is next on my list for a cleanup (after this PR)

ryanlerch commented:
> Found one small issue, when adding karma to update. Karma counter is not changed until refresh.
> ![Screenshot from 2019-09-12 13-42-59](https://user-images.githubusercontent.com/6943409/64781533-59685880-d563-11e9-9b22-1c54b8c18c6b.png)

It looks like this was an existing bug, (i.e. not a regression). Filed #3502 to track this one, and will probably fix it in a seperate PR. (this one is pretty loaded already.)

Zlopez commented:
> 
> hmmm, i didn't touch jquery, and not getting the warning here either. Does the warning also come up when running develop?

I don't see this error on develop branch, it must be introduced by something you added in this PR.
I'm seeing this on Firefox 69.

EDIT: I tried it again today and I don't see the error on this PR anymore. Don't know why it was shown yesterday.

ryanlerch committed:
remove extra font-awesome folder that we don't use

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove old fonts (cantarell and comfortaa) that we dont actually use

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove unused excanvas, flot and jquery

for some reason, we have a jquery here that we dont use either.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove datepicker js, and use the browser's version

the only place we used the datepicker is on the override
form, but now we use the stock html5 version.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove cabbage and konami js

these have not been used since the last refresh of the UI
a few years back. cabbage made a bodhi logo spin in the
background when loading stuff, and konami made the cabbage
spin when the konami code was typed in any page.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove old unused css files

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
reorganize the 3rd party js and css

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
remove old spinner.gif, make all spinners the same

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
move js and css includes into blocks

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
fix integration tests

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch commented:
Just added a bunch of new commits. 

The initial commit is still the same, and does most of the refresh.

The other commits do a clean-out of all the old static files that were not used, or not needed anymore. Since these are removing a lot of files, i kept these seperate, to try to make it easier to review.

The last commit is to fix the broken integration tests.

ryanlerch committed:
try to fix the unit tests on ci

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
fix test_short_false_with_title_sanitizes_unsafe_tags

python-bleach fixed an issue in v3 that "fixed" output on
clean, adding closing tags. This commit changes the test to
test for both output based on the bleach version installed.

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

abompard commented on bodhi/server/static/css/site.css:
```diff
@@ -285,7 +269,7 @@ ul.updateslist li {
   line-height: 1em;
   font-weight: 200;
-}
+}*/
```
Do you want to leave this commented out?

ryanlerch replied:
Nope! this is not needed anymore. removed it entirely in this updated [commit](https://github.com/fedora-infra/bodhi/pull/3500/commits/2e45e91f18ca0609cc1b3e327434bfdabcecb91d)

abompard commented on bodhi/tests/server/services/test_releases.py:
```diff
@@ -480,34 +484,120 @@ def test_get_single_release_html_two_same_updates_same_month(self):
+                        )
+                        self.db.add(update)
+                        self.db.flush()
+                        count = count + 1
```
This function is creating Updates, there's a good chance that this is causing the timeout because creating updates now sends out a Fedora Messaging message, and it needs to be wrapped with mock_sends in the unit tests or the code will just wait a long time to connect to a RabbitMQ server. All this delay piles up and usually causes a timeout.

ryanlerch replied:
Thanks, updated [commit](https://github.com/fedora-infra/bodhi/pull/3500/commits/2e45e91f18ca0609cc1b3e327434bfdabcecb91d) to do this (hopfully correctly)

abompard commented on bodhi/tests/server/test_util.py:
```diff
@@ -33,6 +34,16 @@
+        val = int(val)
+    except ValueError:
+        pass
+    bleach_v[idx] = val
```
Maybe you can use `pkg_resources.parse_version()`

ryanlerch replied:
done, and updated this commit with the change.

https://github.com/fedora-infra/bodhi/pull/3500/commits/751e80ff94681e7c997e509771e284a5a2d178a1

abompard commented on bodhi/tests/server/services/test_releases.py:
```diff
@@ -17,15 +17,19 @@
+from datetime import datetime
 
 import webtest
+from bs4 import BeautifulSoup
```
You should also probably add it to the `tests_require` list in `setup.py`, for coherence.

ryanlerch replied:
fixed in this updated commit:

https://github.com/fedora-infra/bodhi/pull/3500/commits/2e45e91f18ca0609cc1b3e327434bfdabcecb91d

abompard reviewed (approved):
Yep, LGTM, thanks! And thanks for all the cleaning up of old JS & CSS :-)
